_posts/2017/2017-02-20-creating-a-sales-analytics-app-with-ruby-on-rails.md

```diff
@@ -0,0 +1,109 @@
+{% endhighlight %}
+
+and then, iterate the array with all the customers.
+But now, you can use other parameters that the API provides as `page` or `results_per_page`.
+So you can do things like
+
+{% highlight ruby %}
+  epages_shop.customers
+  <Epages::CustomersResponse:0x0000
+    @items = [<Epages::Customer:0x0000>, <Epages::Customer:0x0001>, ..., <Epages::Customer:0x0009>],
+    @page = 1,
+    @results = 42,
+    @results_per_page = 10>
+{% endhighlight %}
+
+or with some parameters
+
+{% highlight ruby %}
+  epages_shop.customers({ page: 2, results_per_page: 10 })
+  <Epages::CustomersResponse:0x0000
+    @items = [<Epages::Customer:0x0010>, <Epages::Customer:0x0011>, ..., <Epages::Customer:0x0019>],
+    @page = 2,
+    @results = 42,
+    @results_per_page = 10>
+{% endhighlight %}
+
+so you have more control on the call and it gets easier to do batch operations.
+We also fixed some bugs and updated some dependencies.
+
+Testing things in the console was fun, but we needed a real life scenario to check that the gem was working correctly and to find missing features.
+Lucky for us, our Product Owner came up with task perfect for this situation.
+We needed to create a Sales Analytics App for the ePages [App Store](https://blog.epages.com/us/2016/05/14/the-epages-app-store-everything-for-your-e-commerce-success/).
+
+## The new app in Rails
+
+One of the things that I like the most about working at ePages is that we have lots of colleagues willing to help us.
+We had mockups, designs and texts for the whole Web App, so we could focus on programming.
+
+For simplicity we mimicked more or less the API models and added a bit of Rails Magic so it's even easier to use.
+
+{% image blog/blog-sales-app-model.png auto center %}
+
+After that, we created all the sections of the app: Dashboard, Sales, Products, and Customers.
+All these parts are bound to two dates (start date and end date) so the merchant can get even more useful data faster.
+
+{% imagebasic blog/blog-sales-app-dashboard.png %} image {% endimagebasic %}
+
+The main purpose of the Dashboard was to have all the basic information about a shop at a glance.
+So we decided to include a graph of sales, the amount of sales and orders, as well as the top products, and last orders.
+
+{% imagebasic blog/blog-sales-app-sales.png %} image {% endimagebasic %}
+
+Then we included three charts on the Sales section.
+The first is the same as in the Dashboard, the second one represents the orders on the selected time frame and the last is the revenue per order, the calculation between the first two graphs.
+We also added the option to export the data to an `xlsx` file so the merchant doesn't need an internet connection to review the sales.
+
+{% imagebasic blog/blog-sales-app-customers.png %} image {% endimagebasic %}
+
+The Customers section is a simple table including all the merchant customers ordered via number of orders or the total amount spent.
```

Comment:
The Customers section is a simple table including all the merchant's customers ordered by number of orders or the total amount spent.